airflow/operators/docker_operator.py

```diff
@@ -198,9 +198,21 @@ def execute(self, context):
         if self.force_pull or len(self.cli.images(name=self.image)) == 0:
             self.log.info('Pulling docker image %s', self.image)
             for l in self.cli.pull(self.image, stream=True):
-                output = json.loads(l.decode('utf-8').strip())
-                if 'status' in output:
-                    self.log.info("%s", output['status'])
+                try:
```

Comment:
I think it would be much simpler to use "decode=True" parameter with pull method. I presume it already does a very good job of converting json into dictionary even in case this is a json array. 

See the https://docker-py.readthedocs.io/en/stable/api.html#docker.api.image.ImageApiMixin.pull  which includes  an example with decode=True.

Reply:
Not sure why it happened, but there’s no issue if I call the docker api directly. I will check if I explicitly put this parameter in docker operator. Thanks

Reply:
From my part, I confirmed that issue came from how docker operator in airflow not the docker api.

Reply:
That's exactly the point. when you use decode=True, you get output as dictionaries already - there is no need to do json.loads(l.decode(... So you do not need try/catch at all and the whole code can be simplified.

Reply:
Do you have any idea why I still get the same issue when I pass the `decode=True`? Thanks

Reply:
If I print them out, it looks like:
```
[2019-04-20 10:11:53,334] {docker_operator.py:188} INFO - Starting docker container from image hello-world
[2019-04-20 10:11:53,352] {docker_operator.py:202} INFO - Pulling docker image hello-world
[2019-04-20 10:11:55,977] {docker_operator.py:204} INFO - b'{"status":"Pulling from library/hello-world","id":"latest"}\r\n'
[2019-04-20 10:11:55,978] {docker_operator.py:208} INFO - Pulling from library/hello-world
[2019-04-20 10:11:56,714] {docker_operator.py:204} INFO - b'{"status":"Pulling fs layer","progressDetail":{},"id":"1b930d010525"}\r\n'
[2019-04-20 10:11:56,715] {docker_operator.py:208} INFO - Pulling fs layer
[2019-04-20 10:11:57,816] {docker_operator.py:204} INFO - b'{"status":"Downloading","progressDetail":{"current":501,"total":977},"progress":"[=========================\\u003e                         ]     501B/977B","id":"1b930d010525"}\r\n'
[2019-04-20 10:11:57,816] {docker_operator.py:208} INFO - Downloading
[2019-04-20 10:11:57,860] {docker_operator.py:204} INFO - b'{"status":"Downloading","progressDetail":{"current":977,"total":977},"progress":"[==================================================\\u003e]     977B/977B","id":"1b930d010525"}\r\n{"status":"Verifying Checksum","progressDetail":{},"id":"1b930d010525"}\r\n'
```

Reply:
What does your code look like after you change to decode? Can you paste it here?

Reply:
I passed it via operator like this:
maybe I get you wrong
```
DockerOperator(
        image='hello-world',
        task_id='test',
        decode=True,
        auto_remove=True)
```

Reply:
Indeed wrong :).

I was talking about passing 'decode=True' parameter in cli.pull() method - then you do not need to do "decode('utf-8')' - it is done internally in the pull method (and likely it handles all the cases). then you get dictionary as result so you can do l['status] directly I believe.